### PR TITLE
[bugfix] fixed jpa expired data clean error

### LINKED ISSUE #2310: [BUG] jpa expired data clean error
### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Current Behavior

delete a half data

### Expected Behavior

truncate table

### Steps To Reproduce

_No response_

### Environment

```markdown
HertzBeat version(s):v1.6.0
```

### Debug logs

 
image

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pwallk**
## What's changed?
fixed #2310

## Checklist

- [x]  I have read the [Contributing Guide](https://hertzbeat.apache.org/docs/community/code_style_and_quality_guide)
- [ ]  I have written the necessary doc or comment.
- [ ]  I have added the necessary unit tests and all cases have passed.

## Add or update API

- [ ] I have added the necessary [e2e tests](https://github.com/apache/hertzbeat/tree/master/e2e) and all cases have passed.

### CODE REVIEW

File: warehouse/src/main/java/org/apache/hertzbeat/warehouse/dao/HistoryDao.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public interface HistoryDao extends JpaRepository<History, Long>, JpaSpecificati
      */
     @Modifying
     @Transactional(rollbackFor = Exception.class)
-    @Query(value = "delete from hzb_history limit :delNum", nativeQuery = true)
+    @Query(value = "DELETE FROM hzb_history WHERE id IN ( SELECT t2.id from (SELECT t1.id FROM hzb_history t1 LIMIT ?1) as t2)", nativeQuery = true)
```
Comment (tomsun28) [inline warehouse/src/main/java/org/apache/hertzbeat/warehouse/dao/HistoryDao.java:49]: hi, can you describe the reason change? The `limit xx` is avaiable in postgresql and mysql.

**Comment by pwallk** [comment]
First, the argument cannot be passed correctly and should be changed instead：,secondly postgresql does not support followed by ，Therefore, it should be deleted with nested subqueries (earlier MySQL versions did not support nested subqueries, such as 5.7, 5.6, so two layers of nesting are used).delNumdelete from hzb_history limit ?1deletelimit

**Comment by pwallk** [comment]
This is version 1.6 that I deployed on the server：
Clip_2024-07-22_08-53-12
postgresql throws an exception when the argument can be passed correctly：
Clip_2024-07-22_09-05-50

**Comment by pwallk** [comment]
sorry my review is pending

**Review by tomsun28 — APPROVED** [review]
lgtm 👍👍

**Pull request merged**